File: DentOffice.Model/Termin.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace DentOffice.Model
{
    public class Termin
    {
        public int TerminID { get; set; }
        public string Razlog { get; set; }
        public bool Hitno { get; set; }
        public bool Odobreno { get; set; }
        public DateTime DatumVrijeme { get; set; }

        public int PacijentId { get; set; }
        public Pacijent Pacijent { get; set; }
        public int UslugaId { get; set; }
        public Usluga Usluga { get; set; }
        public string UslugaNaziv { get; set; }

        public string Opis => $"{DatumVrijeme} ({(Odobreno ? "Odobren" : "Neodobren")})";
    }
}

File: dentofficemobile/lib/models/termin.dart
// -- termin.dart --
import 'package:intl/intl.dart';
import 'package:json_annotation/json_annotation.dart';

import 'pacijent.dart';
import 'usluga.dart';

part 'termin.g.dart';

@JsonSerializable()
class Termin {
   int? terminID;
   String razlog;
   bool hitno;
   bool? odobreno;
   bool naCekanju;
   DateTime datumVrijeme;
   int pacijentId;
   Pacijent? pacijent;
   int uslugaId;
   Usluga? usluga;
   String? uslugaNaziv;

   String get opis => "${DateFormat("dd-MM-yyyy").format(datumVrijeme)} (${(odobreno != null && odobreno!) ? "Odobren" : "Neodobren"})";

   Termin({this.terminID,required this.razlog,this.hitno=false,this.odobreno,required this.naCekanju,required this.datumVrijeme,required this.pacijentId, this.pacijent,required this.uslugaId,this.usluga,this.uslugaNaziv,});

   factory Termin.fromJson(Map<String, dynamic> json) => _$TerminFromJson(json);

   Map<String, dynamic> toJson() => _$TerminToJson(this);
}